libs/sdk-ui-dashboard/src/presentation/widget/common/configuration/AttributeFilterConfiguration.tsx

```diff
@@ -20,20 +29,58 @@ export const AttributeFilterConfiguration: React.FC<IAttributeFilterConfiguratio
     const dfMap = useDashboardSelector(selectAttributeFilterDisplayFormsMap);
     const attrMap = useDashboardSelector(selectAllCatalogAttributesMap);
 
+    const displayForms = useMemo(() => {
+        return attributeFilters.map((filter) => filter.attributeFilter.displayForm);
+    }, [attributeFilters]);
+
+    const {
+        status,
+        run: getAttributes,
+        result: attributes,
+    } = useDashboardQueryProcessing<
+        QueryAttributeByDisplayForm,
+        IAttributeMetadataObject[],
+        Parameters<typeof queryAttributeByDisplayForm>
+    >({
+        queryCreator: queryAttributeByDisplayForm,
+    });
+
+    useEffect(() => {
+        getAttributes(displayForms);
+    }, [displayForms, getAttributes]);
+
+    const isLoading = status === "pending" || status === "running";
+
+    if (isLoading) {
+        return <span className={"gd-spinner small s-attribute-filter-configuration-loading"} />;
+    }
+
+    if (!attributes) {
+        return null;
+    }
+
     return (
-        <div>
+        <div className="s-attribute-filter-configuration">
             {attributeFilters.map((filter) => {
                 const displayForm = dfMap.get(filter.attributeFilter.displayForm);
                 invariant(displayForm, "Inconsistent state in AttributeFilterConfiguration");
 
-                const attribute = attrMap.get(displayForm.attribute);
+                const findAttributeByDisplayForm = attributes.find((attribute) =>
```

Comment:
This logic also do some mess in this render could you separate

Reply:
Added util function for this.